Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -5891,6 +5891,11 @@ void LocalFrameView::fireLayoutRelatedMilestonesIfNeeded()
             addPaintPendingMilestones(DidFirstMeaningfulPaint);
             if (requestedMilestones & DidFirstVisuallyNonEmptyLayout)
                 milestonesAchieved.add(DidFirstVisuallyNonEmptyLayout);
+
+            if (m_frame->isMainFrame()) {
+                if (auto* page = m_frame->page())
+                    page->didFirstMeaningfulPaint();
```

Comment:
Not quite right, as discussed on Slack.

Reply:
Indeed — this should go into `firePaintRelatedMilestonesIfNeeded()`, when we actually finish the meaningful paint.